### ISSUE TITLE
switch to a new repo

### ISSUE THREAD

**Issue by 7flash** [jesseduffield/lazygit#3947]
Currently I can press Ctrl+R to switch into one of recently open respositories, but I cannot to switch into any new repository, unfortunately. Even if I would do lazygit manually in another tab inside of that repo, it still does not appear in the list of "Recent repositoreis"

**Comment by mark2185** [comment]
Yeah, it looks like we don't reload the state when showing the list of recent repositories. At least the workaround is easy, quit and launch `lazygit` again.

I guess this could be a good first issue if someone wants to implement this, but I'm not entirely convinced we need it.

**Comment by 7flash** [comment]
What is better recommended way to add a new repo? Maybe a new hotkey "n" can be introduced under Status tab in similar way as it works for adding new origin under Remotes tab?

**Comment by mark2185** [comment]
How often do you add new repos vs relaunch `lazygit`?

I usually have 1 `lazygit` instance per repo so I rarely use the switch repo feature.

**Comment by 7flash** [comment]
> How often do you add new repos vs relaunch `lazygit`?
> 
> 
> 
> I usually have 1 `lazygit` instance per repo so I rarely use the switch repo feature.

I use a single lazygit instance running in its own dedicated terminal so every time I'll start working in a new repo I'd have to relaunch it and I think it would be more consistent behavior with origins and branches if I could just add a new path with "n"

**Comment by mark2185** [comment]
> I think it would be more consistent behavior with origins and branches if I could just add a new path with "n"

Probably, but I'm not sure it warrants the complexity. It would probably be better to just reload the cache when opening up a list of recent repos.

**Comment by rafgugi** [comment]
how about making this behavior configurable? users could choose between caching the repo list (current behavior) or reloading it from the config file each time the list is opened. wdyt?
